processor/workers.go

```diff
@@ -260,6 +262,11 @@ func CountStats(fileJob *FileJob) {
 		return
 	}
 
+	// If we have multiple languages then we need to guess which one this might be
+	if len(fileJob.Languages) != 1 {
```

Comment:
I know it _shouldn't_ happen, but I'd still be more defensive here of `len == 0`. Slices can also be `nil`, but that might be a safer / more reasonable assumption.

Reply:
Sorry not following. You mean check if it is empty? IE

```
if len(fileJob.Languages) == 0 {
```

Reply:
Yeah, that's what I meant.

Reply:
Refactored it out totally. Changed method name to `determineLanguage` and gave it a quick out. This was done to accommodate removal of setting `Language` based on `PossibleLanguages`